### PR TITLE
Add versions to curated plugin response

### PR DESCRIPTION
**Pull request by mfridman**
This PR adds a `versions` field to the curated plugin response so we can display the requested plugin as well as a versions list. 

An argument could be made that we should have a separate endpoint to retrieve all the versions, probably utilizing a List endpoint and then doing filtering. But I think we should avoid proliferation of distinct RPCs and solve consumer objectives.

### CODE REVIEW

**Review by renahlee — APPROVED** [review]

File: proto/buf/alpha/registry/v1alpha1/plugin_curation.proto [hunk 1]
```diff
@@ -218,6 +218,13 @@ message GetLatestCuratedPluginRequest {
 }
 
 message GetLatestCuratedPluginResponse {
```
Comment (unmultimedio) [inline proto/buf/alpha/registry/v1alpha1/plugin_curation.proto:220]: I guess this and L116 are not true anymore then?

```suggestion
message GetCuratedPluginResponse {
```

Although this'd be a breaking change. So may be just a clarification point and not an actual suggestion.

File: proto/buf/alpha/registry/v1alpha1/plugin_curation.proto [hunk 2]
```diff
@@ -218,6 +218,13 @@ message GetLatestCuratedPluginRequest {
...
-  // The latest plugin.
   CuratedPlugin plugin = 1;
+  // versions is a semver-sorted list in ascending order.
```
Comment (unmultimedio) [inline proto/buf/alpha/registry/v1alpha1/plugin_curation.proto:222]: I suggest showing latest first, when checking versions we usually see them from the most recent to the oldest. Same for the revisions.

File: proto/buf/alpha/registry/v1alpha1/plugin_curation.proto (hunk 1 above)
Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/plugin_curation.proto:220]: If you omit the version in the request, then it does return the latest curated plugin. And if you include the version, then it'll return the latest curated plugin by revision.

We might want to add revision as an optional, but that hasn't come up yet.

Comment (unmultimedio) [reply proto/buf/alpha/registry/v1alpha1/plugin_curation.proto:220]: Gotcha, thanks for the clarification :)

File: proto/buf/alpha/registry/v1alpha1/plugin_curation.proto (hunk 2 above)
Comment (mfridman) [reply proto/buf/alpha/registry/v1alpha1/plugin_curation.proto:222]: Fair, updated.

**Review by unmultimedio — APPROVED** [review]

**Pull request merged**